File: C++/Solution73.cpp
#include<iostream>
#include<vector>
#include<algorithm>
#include<unordered_map>
#include<unordered_set>
using namespace std;

class Solution73 {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        unordered_set<int> row, col;
        for(int i = 0; i < matrix.size(); i++){
            for(int j = 0; j < matrix[0].size(); j++){
                if(matrix[i][j] == 0){
                    row.insert(i);
                    col.insert(j);
                }
            }
        }

        for(auto ele: row){
            for(int j = 0; j < matrix[0].size(); j++){
                matrix[ele][j] = 0;
            }
        }
        for(auto ele: col){
            for(int i = 0; i < matrix.size(); i++){
                matrix[i][ele] = 0;
            }
        }
    }
};

File: Java/other/Solution73.java
import java.util.Set;
import java.util.HashSet;

class Solution73 {
    public void setZeroes(int[][] matrix) {
    	Set<Integer> row = new HashSet<>();
    	Set<Integer> col = new HashSet<>();
    	for(int i = 0; i < matrix.length; i++){
    		for(int j = 0; j < matrix[0].length; j++){
    			if(matrix[i][j] == 0){
    				if(!row.contains(i)){
    					row.add(i);
    				}
    				if(!col.contains(j)){
    					col.add(j);
    				}
    			}
    		}
    	}
    	for(Integer x: row){
    		for(int j = 0; j < matrix[0].length; j++){
    			matrix[x][j] = 0;
    		}
    	}
    	for(Integer y: col){
    		for(int j = 0; j < matrix.length; j++){
    			matrix[j][y] = 0;
    		}
    	}
    }
}